### PR TITLE
Changed the implementation of the logger from winston to pino

### LINKED ISSUE #676: Look into winston logger resource usage
### System Details
- **Papiea Engine version**: 0.9.46
- **Papiea SDK version:** same
- **Papiea Client version:** same

### Describe the Bug

Customer-1 has periodically seen OOM crashes in Node. Upon inspection of core dumps with `strings`, the most common string found was a log tag of the form `<provider>/<version>/...`, which makes me think there is some issue with how we are using the Winston logger in the TypeScript SDK.

Not sure if this is also related to the EventEmitter leaks we have been seeing over time, but I suspect it might be. 

### Bug information
Let's look into how we are using the winston logger to see if it needs to be explicitly closed once we are no longer using it. If so, we probably need to update the SDK to handle this.

### Steps to Reproduce
1. Leave a TypeScript Papiea provider running for long enough
2. Eventually, Node crashes with an out-of-memory error

<hr>
<em>Readers: If you are also experiencing this bug, please vote for it by giving the ORIGINAL POST a thumbs-up using the :smiley: button.  You are welcome to leave comments and discuss the issue if you have more details to add, but "Me too!" comments are not counted by the voting system.</em>

### PR DESCRIPTION
**Pull request by Emulebest**
resolves #676

### CODE REVIEW

File: papiea-backend-utils/package.json [hunk 1]
```diff
@@ -16,11 +16,14 @@
...
     "@types/jaeger-client": "^3.15.4",
+    "@types/pino": "^6.3.8",
     "express": "^4.17.1",
     "jaeger-client": "^3.18.1",
     "logform": "^2.1.2",
     "opentracing": "^0.14.5",
     "papiea-core": "file:../papiea-core",
+    "pino": "^6.11.3",
+    "pino-pretty": "^4.8.0",
     "winston": "^3.3.3"
```
Comment (joshua-berry-ntnx) [inline papiea-backend-utils/package.json:27]: Should we remove winston?

File: papiea-backend-utils/src/logging.ts [hunk 2]
```diff
@@ -45,17 +42,13 @@ export interface Logger {
 }
 
 export class LoggerHandle {
-    constructor(private logger: winston.Logger) {
-    }
-
-    cleanup(): void {
-        this.logger.close()
-        this.logger.clear()
```
Comment (joshua-berry-ntnx) [inline papiea-backend-utils/src/logging.ts:53]: Does pino still need some kind of cleanup to be done?

File: papiea-backend-utils/src/logging.ts [hunk 3]
```diff
@@ -11,15 +10,15 @@ export interface LoggingVerbosityOptions {
...
     audit: 4,
-    warn: 5,
-    notice: 6,
-    info: 7,
-    debug: 8,
+    warn: 3,
+    notice: 2,
+    info: 1,
+    debug: 0,
 }
```
Comment (joshua-berry-ntnx) [inline papiea-backend-utils/src/logging.ts:22]: Do we still need this if we're not exposing a generic log() function anymore?

**Review by joshua-berry-ntnx — APPROVED** [review]
Mostly looks good; a couple questions below.

File: papiea-engine/src/auth/casbin.ts [hunk 4]
```diff
@@ -36,7 +36,7 @@ export class CasbinAuthorizer extends Authorizer {
                 throw new PermissionDeniedError({ message: `User does not have permission for the entity on provider: ${provider?.prefix}/${provider?.version}. Add policy to allow access to the user.`, entity_info: { provider_prefix: provider?.prefix, provider_version: provider?.version, additional_info: { "user": JSON.stringify(user), "action": action, "entity": JSON.stringify(object) }}});
             }
         } catch (e) {
-            this.logger.error("CasbinAuthorizer checkPermission error", e);
+            this.logger.error({err: "CasbinAuthorizer checkPermission error", cause: e});
```
[the remark this answers is not in the document]
Comment (Emulebest) [reply papiea-engine/src/auth/casbin.ts:39]: Pino could take both object and string

File: papiea-backend-utils/src/logging.ts (hunk 3 above)
Comment (Emulebest) [reply papiea-backend-utils/src/logging.ts:22]: Yes, we need this

File: papiea-backend-utils/src/logging.ts (hunk 2 above)
Comment (Emulebest) [reply papiea-backend-utils/src/logging.ts:53]: Nope

File: papiea-backend-utils/src/logging.ts [hunk 5]
```diff
@@ -94,23 +87,31 @@ export class LoggerFactory {
...
         }
 
+        let custom_logger: ((object: object) => object) | undefined = undefined
+
         switch (opts.pretty_print) {
         case false: // json
-            formatArgs.push(winston.format.json())
             break
 
         case true:  // pretty
```
[the remark this answers is not in the document]
Comment (Emulebest) [reply papiea-backend-utils/src/logging.ts:109]: I'd still keep it as an option to adjust later, this seems like a very generic ask for any logger

**Pull request merged**